torch/_dynamo/symbolic_convert.py

```diff
@@ -3981,8 +3981,25 @@ def __init__(
         f_builtins = f_globals["__builtins__"]
         if not isinstance(f_builtins, dict):
             f_builtins = f_builtins.__dict__
-        instructions = cleaned_instructions(code)
-        propagate_line_nums(instructions)
+
+        # Get the cached instructions. These instructions are safe to cache
+        # because we dont mutate them in transform_code_object (those
+        # instructions are for the top most Instruction translator).  Also, we
+        # have to be careful about not using _cached_cleaned_instructions here
+        # because that function is global, while we want the the cache to be
+        # alive only during a compmilation.
+        tracing_ctx = parent.output.tracing_context
+        instructions = None
+        if tracing_ctx:
+            if tracing_ctx.previously_cleaned_instructions.get(code):
+                instructions = tracing_ctx.previously_cleaned_instructions[code]
+
+        if instructions is None:
+            instructions = cleaned_instructions(code)
+            propagate_line_nums(instructions)
```

Comment:
Looks like this basically
1. Go from a global cache (from `_cached_cleaned_instructions `) for the code object, to a `tracing_context`-local cache?
2. Avoids a `propagate_line_nums` call if `code` was cached.

So is (1) mainly to enable (2), which then saves compilation time?